packages/audiofileplayer/android/src/main/java/com/google/flutter/plugins/audiofileplayer/AudiofileplayerPlugin.java

```diff
@@ -112,37 +121,103 @@
   // Used when defining an Intent from a custom media button.
   public static final String CUSTOM_MEDIA_BUTTON_EXTRA_KEY = "customMediaButton";
 
-  private final Registrar registrar;
-  private final Map<String, ManagedMediaPlayer> mediaPlayers;
-  private final MethodChannel methodChannel;
+  private Registrar registrar;
+  private Activity activity;
+  private Map<String, ManagedMediaPlayer> mediaPlayers;
+  private MethodChannel methodChannel;
+  private Context context;
+  private ActivityPluginBinding activityPluginBinding;
 
   private MediaBrowserCompat mediaBrowser;
   private MediaControllerCompat mediaController;
 
   public static void registerWith(Registrar registrar) {
-    final MethodChannel methodChannel = new MethodChannel(registrar.messenger(), CHANNEL);
-    final AudiofileplayerPlugin instance = new AudiofileplayerPlugin(registrar, methodChannel);
-    methodChannel.setMethodCallHandler(instance);
+    AudiofileplayerPlugin instance = new AudiofileplayerPlugin();
+    instance.registrar = registrar;
+    instance.activity = registrar.activity();
```

Comment:
If the app is registered in the background, you won't get activity here. For v1, you will need to keep the registrar and always get activity from registrar.